homeassistant/components/elmax/common.py

```diff
@@ -114,21 +126,21 @@ async def _async_update_data(self):
             _LOGGER.error("Refused username/password")
             raise ConfigEntryAuthFailed from err
         except ElmaxApiError as err:
-            raise HomeAssistantError(
-                f"Error communicating with ELMAX API: {err}"
-            ) from err
+            raise UpdateFailed(f"Error communicating with ELMAX API: {err}") from err
         except ElmaxNetworkError as err:
-            raise HomeAssistantError(
+            raise UpdateFailed(
                 "Network error occurred while contacting ELMAX cloud"
             ) from err
         except Exception as err:
```

Comment:
We shouldn't catch broad exception. This is already handled by the coordinator.

https://github.com/home-assistant/core/blob/8df0bc9d57c0b0e2c987e4ccaab2d392e9dcf88e/homeassistant/helpers/update_coordinator.py#L242-L247

Reply:
DONE.
Didn't notice that: I agree we can leverage the update_coordinator error handling.